addons/website_event_booth/static/src/js/booth_register.js

```diff
@@ -16,11 +16,18 @@ publicWidget.registry.boothRegistration = publicWidget.Widget.extend({
     },
 
     start() {
-        this.eventId = parseInt(this.$el.data('event-id'));
+        this.eventId = parseInt(this.el.dataset.eventId);
         this.activeType = false;
         this.boothCache = {};
+        this.boothsFirstRendering = true;
+        this.selectedBoothIds = [];
         return this._super.apply(this, arguments).then(() => {
-            this.$('input[name="booth_category_id"]:enabled:first').click();
+            const boothSelectedCategory = this.el.querySelector('.o_wbooth_selected_category');
+            if (boothSelectedCategory) {
+                // A single id is interpreted as a number. Hence the toString method.
+                this.selectedBoothIds = this.el.querySelector('.o_wbooth_booths').dataset.selectedBoothIds.toString().split(',');
+                boothSelectedCategory.click();
```

Comment:
I see that we still have some post-processing, is that part complex to do in the server-rendered template?